app/routes.py

```diff
@@ -0,0 +1,381 @@
+
+        if "name" not in put_request_body:
+            return jsonify(None), 400
+
+        customer.name=put_request_body["name"]
+        customer.phone=put_request_body["phone"]
+        customer.postal_code=put_request_body["postal_code"]
+
+        db.session.commit()
+
+        updated_customer_response = {
+            "name": customer.name,
+            "phone": customer.phone,
+            "postal_code": customer.postal_code
+        }
+
+        return jsonify(updated_customer_response), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(customer)
+        db.session.commit()
+
+        if customer == []:
+            cust_invalid_delete_response = {"message": f"Customer {customer_id} was not found"}
+
+            return jsonify(cust_invalid_delete_response), 404
+        else:
+            cust_delete_response = customer.get_cust_dict()
+
+            return jsonify(cust_delete_response), 200
+
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"])
+def videos_checked_out(customer_id):
+    # return list of videos a customer currently has checked out
+    if not customer_id.isnumeric():
+        return jsonify(None), 400
+
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return jsonify({"message": f"Customer {customer_id} was not found"}), 404
+
+    results = db.session.query(Video, Rental).filter(Rental.customer_id == customer.id, Rental.checked_out == True).all()
+
+    video_list = [
+        {
+            "release_date": video.release_date,
+            "title": video.title,
+            "due_date": rental.due_date,
+        } 
+        for video, rental in results
+    ]
+
+    if video_list == []:
+        return jsonify(video_list), 200
+
+    return jsonify(video_list), 200
+    
+
+# RENTAL ROUTES
+# CHECK-OUT ENDPOINT
+@rentals_bp.route("/check-out", methods=["POST"])
+def handle_rental_check_outs():
+    rentals_request_body = request.get_json()
+    
+    if "customer_id" not in rentals_request_body or "video_id" not in rentals_request_body:
+        return jsonify(None), 400
+
+    video = Video.query.get(rentals_request_body["video_id"])
+    customer = Customer.query.get(rentals_request_body["customer_id"])
+
+    if video is None or customer is None:
+        return jsonify(None), 404
+
+    videos_checked_out = Rental.query.filter(Rental.video_id == rentals_request_body["video_id"], Rental.checked_out == True).count()
+
+    if videos_checked_out == video.total_inventory:
+        return jsonify({"message": "Could not perform checkout"}), 400
+
+    new_rental = Rental(
+        customer_id=customer.id,
+        video_id=video.id,
+        checked_out=True
+    )
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    videos_checked_out = Rental.query.filter(Rental.video_id == rentals_request_body["video_id"], Rental.checked_out == True).count()
+
+    avail_inventory = video.total_inventory - videos_checked_out
```

Comment:
you could add videos_checkout_count to your model then subtract from the count by doing` rental.videos_checkout_count +=1 ` and add total inventory would be `video.total_inventory -=1`